packages/flutter/lib/src/material/scrollbar.dart

```diff
@@ -110,6 +111,13 @@ class Scrollbar extends StatelessWidget {
   /// null, the default value is 12.0 pixels.
   final double? hoverThickness;
 
+  /// Controls if the leading and trailing buttons will show.
+  ///
+  /// If this property is null, then [ScrollbarThemeData.buttonsVisibility] of
+  /// [ThemeData.scrollbarTheme] is used. If that is also null, the buttons will
+  /// not show by default.
+  final MaterialStateProperty<bool?>? buttonsVisibility;
```

Comment:
So this is a bit out there, what if the user wanted consolidated buttons? Like grouped together at one end of the track or another? This was a use case I came across way back when I experimented with this.

![image](https://user-images.githubusercontent.com/16964204/141368553-ef196ae7-d4e2-4764-899e-e8bc9e4beb48.png)


May this could be an enum instead of a bool?

```dart
enum ScrollbarButtonStyle {
  // no buttons
  none,
  // the current style here
  split,
  // these are the consolidated ones
  groupedLeading,
  groupedTrailing,
}

Reply:
This is the first time I have seen this style of bar, but I think it is really useful, I try to achieve it.